boot/bootutil/src/image_rsa.c

```diff
@@ -71,6 +106,157 @@ bootutil_parse_rsakey(mbedtls_rsa_context *ctx, uint8_t **p, uint8_t *end)
     return 0;
 }
 
+#if !MYNEWT_VAL(BOOTUTIL_RSA_PKCS1_15)
+/*
+ * Compute the RSA-PSS mask-generation function.  Assumptions are that
+ * the mask length will be less than 256 * PSS_HLEN, and therefore we
+ * never need to increment anything other than the low byte of the
+ * counter.
+ */
+static void
+pss_mgf1(uint8_t *mask, const uint8_t *hash)
+{
+    bootutil_sha256_context ctx;
+    uint8_t counter[4] = { 0, 0, 0, 0 };
+    uint8_t htmp[PSS_HLEN];
+    int count = PSS_MASK_LEN;
+    int bytes;
+
+    while (count > 0) {
+        bootutil_sha256_init(&ctx);
+        bootutil_sha256_update(&ctx, hash, PSS_HLEN);
+        bootutil_sha256_update(&ctx, counter, 4);
+        bootutil_sha256_finish(&ctx, htmp);
+
+        counter[3]++;
+
+        bytes = PSS_HLEN;
+        if (bytes > count)
+            bytes = count;
+
+        memcpy(mask, htmp, bytes);
+        mask += bytes;
+        count -= bytes;
+    }
+}
+
+/*
+ * Validate an RSA signature, using RSA-PSS, as described in PKCS #1
+ * v2.2, section 9.1.2, with many parameters required to have fixed
+ * values.
+ */
+static int
+bootutil_cmp_rsasig(mbedtls_rsa_context *ctx, uint8_t *hash, uint32_t hlen,
+  uint8_t *sig)
+{
+    bootutil_sha256_context shactx;
+    uint8_t buf[MBEDTLS_MPI_MAX_SIZE];
```

Comment:
So `buf` here is the `DB`, so I suggest that you rename the variable so it's easier to follow what is going on when comparing this with specs etc.